cellxgene_resources/curation_qa.ipynb

```diff
@@ -656,6 +656,133 @@
     "    ][['donor_id','development_stage_ontology_term_id']]"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Check donor sex metadata against raw gene expression counts (genes found in PAR and NonPAR regions)**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Read in par regions\n",
+    "par_file = 'sex_analysis_regions/list_chrY_PAR_genes.txt'\n",
+    "nonpar_file =  'sex_analysis_regions/list_chrY_nonPAR_genes.txt'\n",
+    "par = pd.read_table(par_file, names=['par_genes'])\n",
+    "nonpar = pd.read_table(nonpar_file, names=['nonpar_genes'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Use raw matrix counts for sex analysis\n",
+    "raw_adata = ad.AnnData(adata.raw.X, var=adata.raw.var, obs=adata.obs)  # Create raw anndata if adata.raw exists otherwise raw_adata = adata\n",
```

Comment:
We'll probably also want to explore a way to do this without creating a whole other AnnData object. I can imagine some of the bigger Datasets choking at this point.

Reply:
> not all Datasets will have raw.X. Will need to account for those that don't (they'll have raw counts in .X)

i have this in the comment, but maybe it should be a separate line of code (commented out)?

Reply:
Look elsewhere in the notebook, you'll see several places where the logic is built in so the user doesn't need to pay attention `if adata.raw:`...`else:`...

Reply:
>We'll probably also want to explore a way to do this without creating a whole other AnnData object. I can imagine some of the bigger Datasets choking at this point.

Looking at Jenny's code from CXG-243, that code creates a new AnnData, but subset to only the genes of interest, so it's a much smaller object `adata = adata[:,adata.var['feature_id'].isin(male_genes+female_gene)]`
(but in our case, use a new variable name so that we don't overwrite the main adata object)
It looks like her ntbk may also have some ideas for more succinctly calculating the mean expression

Reply:
Sorry if I'm not understanding correctly, but creating the smaller anndata objects is what i'm doing in these two lines [685/686](https://github.com/Lattice-Data/lattice-tools/pull/124/files#diff-c8c884b4f0b3b0bae8767087ebf2f17847330968c6337f01b163d4ff8eb9b597:~:text=%22par_adata%20%3D%20raw_adata%5B%3A%2Craw_adata,gene_ids%27%5D). 

I created a `raw_adata` variable to account for datasets that have the raw counts in `.X`, this method is faster than reusing the subsetting code for either `raw.X` or `.X`. 
```
if adata.raw:
    par_adata = adata.raw[:,adata.raw.var.index.isin(par_ids['gene_ids'])]
    nonpar_adata = adata.raw[:,adata.raw.var.index.isin(nonpar_ids['gene_ids'])] 

else:
    par_adata = adata[:,adata.var.index.isin(par_ids['gene_ids'])]  
    nonpar_adata = adata[:,adata.var.index.isin(nonpar_ids['gene_ids'])] 
```
Comparing the current code vs the above code using various datasets, the run time is longer using the code above as the datasets get larger. For example with a 5GB dataset, Elapsed time: 0.40940117835998535 seconds for current code vs Elapsed time: 1.7622332572937012 seconds for code above.